packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,266 @@
+#
+# ------------------------------------------------------------------------------
+
+"""HTTP connection and channel."""
+
+import asyncio
+import logging
+import threading
+import time
+import json
+import yaml
+from asyncio import CancelledError
+from threading import Thread
+from typing import Any, Dict, List, Optional, Set, cast
+
+from fetch.p2p.api.http_calls import HTTPCalls
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.helpers.base import locate
+from aea.mail.base import AEAConnectionError, Address, Envelope, EnvelopeContext
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPChannel:
+    """A wrapper for an RESTful API."""
+
+    def __init__(
+        self,
+        address: Address,
+        api_spec: str,
+        provider_addr: str,
+        provider_port: int,
+        excluded_protocols: Optional[Set[PublicId]] = None,
+    ):
+        """
+        Initialize a channel.
+
+        :param address: the address
+        """
+        self.address = address
+        self.provider_addr = provider_addr
+        self.provider_port = provider_port
+        self.in_queue = None  # type: Optional[asyncio.Queue]
+        self.loop = None  # type: Optional[asyncio.AbstractEventLoop]
+        self._httpCall = None  # type: Optional[HTTPCalls]
+        self.excluded_protocols = excluded_protocols
+        self.thread = Thread(target=self.receiving_loop)
+        self.lock = threading.Lock()
+        self.stopped = True
+        logger.info("Initialised the http channel")
+
+    def connect(self):
+        """
+        Connect.
+
+        :return: an asynchronous queue, that constitutes the communication channel.
+        """
+        with self.lock:
+            if self.stopped:
+                self._httpCall = HTTPCalls(
+                    server_address=self.provider_addr, port=self.provider_port
+                )
+                self.stopped = False
+                self.thread.start()
+                logger.debug("HTTP Channel is connected.")
+                self.try_register()
+
+    def send(self, envelope: Envelope) -> None:
+        """
+        Process the envelopes.
+
+        :param envelope: the envelope
+        :return: None
+        """
+        assert self._httpCall is not None
+
+        if self.excluded_protocols is not None:
+            if envelope.protocol_id in self.excluded_protocols:
+                logger.error(
+                    "This envelope cannot be sent with the oef connection: protocol_id={}".format(
+                        envelope.protocol_id
+                    )
+                )
+                raise ValueError("Cannot send message.")
+
+        self._httpCall.send_message(
+            sender_address=envelope.sender,
+            receiver_address=envelope.to,
+            protocol=str(envelope.protocol_id),
+            context=b"None",
+            payload=envelope.message,
+        )
```

Comment:
Keep in mind that if you are using the p2p library from fetch is specific for the p2p implementation that we have in the Ledger constelation. The message in the library is in the form of :
``` 
 msg = {FROM: {NODE_ADDRESS: self.node_address,
                      SENDER_ADDRESS: send_address},
               TO: {NODE_ADDRESS: self.node_address,
                    RECEIVER_ADDRESS: receive_address},
               PROTOCOL: protocol,
               CONTEXT: context,
               PAYLOAD: payload}
```
I mean, I am not sure if this library is useful for what you are doing :)

Reply:
Agreed, the P2P connection has very little relevance here